### PR TITLE
Improvement: rename the variables in TCP reconnect function and some default constants

### PR DESCRIPTION
**Pull request by No-SilverBullet**
**What this PR does**:

**rename the variables in TCP reconnect function and some default constants.**

1.rename constant 'reconnectInterval' to defaultReconnectInterval and 'maxTimes' to defaultMaxReconnectAttempts, indicates that these two constants are the default settings.

2.rename vars in reConnect():
- num->sessionNum,indicates the number of current sessions.
- max->maxReconnectAttempts,indicates the maximum attempt times of reconnections.
- times->reconnectAttempts, indicates the current number of reconnect attempts.
- interval->reconnectInterval,indicates the interval between each reconnection.
- maxDuration->maxReconnectInterval, indicates the max interval between each reconnection.

**Which issue(s) this PR fixes**:
NONE

**Special notes for your reviewer**:
NONE
**Does this PR introduce a user-facing change?**:

```release-note
Add new client-side attribute 'maxReconnectAttempts' to represent the max reconnect attempts in maintaining the tcp connetion pool size, the default value is setted to 100.
```

## Summary by CodeRabbit

- **New Features**
  - Introduced a configurable maximum number of TCP reconnect attempts to improve connection reliability.

- **Improvements**
  - Enhanced error handling during connection read and write deadline settings for better debugging and stability.

### CODE REVIEW

**Review by AlexStocks — APPROVED** [review]

File: transport/options.go [hunk 1]
```diff
@@ -168,3 +168,12 @@ func WithClientTlsConfigBuilder(tlsConfigBuilder TlsConfigBuilder) ClientOption
...
 }
+
+// WithReconnectAttempts @maxReconnectAttempts is max reconnect attempts.
+func WithReconnectAttempts(maxReconnectAttempts int) ClientOption {
+	return func(o *ClientOptions) {
+		if 0 < maxReconnectAttempts {
+			o.maxReconnectAttempts = maxReconnectAttempts
+		}
+	}
+}
```
Comment (Lvnszn) [inline transport/options.go:179]: 这段代码并没有被使用，可以先不加或者加注释。否则被人调用之后发现没有效果反而觉得是个问题。

**Comment by Lvnszn** [comment]
@AlexStocks 雨哥，看着 reconnect 那边的逻辑其实可以加个 issue，太多参数都是临时变量，不利于维护。其实变成 client 里面的参数会更好一点。

File: transport/options.go (hunk 1 above)
Comment (AlexStocks) [reply transport/options.go:179]: > 这段代码并没有被使用，可以先不加或者加注释。否则被人调用之后发现没有效果反而觉得是个问题。

这是给上面提供的接口。要不 @No-SilverBullet 你在代码示例里面调用下？

File: transport/client.go [hunk 2]
```diff
@@ -41,10 +41,11 @@ import (
...
 
 const (
-	reconnectInterval = 3e8 // 300ms
-	connectInterval   = 5e8 // 500ms
-	connectTimeout    = 3e9
-	maxTimes          = 10
+	defaultReconnectInterval    = 3e8 // 300ms
+	connectInterval             = 5e8 // 500ms
+	connectTimeout              = 3e9
+	defaultMaxReconnectAttempts = 100
```
Comment (Lvnszn) [inline transport/client.go]: 这个要不调小一点，看着是短时间内发起 100次重连请求。

**Review by Lvnszn — APPROVED** [review]

File: transport/options.go (hunk 1 above)
Comment (No-SilverBullet) [reply transport/options.go:179]: 11

**Comment by No-SilverBullet** [comment]
> @AlexStocks 雨哥，看着 reconnect 那边的逻辑其实可以加个 issue，太多参数都是临时变量，不利于维护。其实变成 client 里面的参数会更好一点。

我把里面的一些变量调整为短变量申明的方式，这样看上去可能会好一点。然后除了maxReconnectAttempts（这个参数是client-side需要关心的参数，其他都不需要放到client里）

**Review by AlexStocks — APPROVED** [review]

**Review by Lvnszn — APPROVED** [review]

**Pull request merged**